components/chatbot/controller.js

```diff
@@ -155,8 +118,12 @@ function Controller() {
             source: sourceStatus,
           };
 
-          getChatTitle(chatId);
-          console.log("this is finalBot Message", finalBotMessage);
+          // The chatId has already changed before the fetch completed when the user switched to another chat.
+          if (chatId !== getCurrentChatId()) {
```

Comment:
The chatId has already changed before the fetch completed when the user switched to another chat.